superset/viz.py

```diff
@@ -491,6 +491,18 @@ def data(self):
     def get_csv(self):
         df = self.get_df()
         include_index = not isinstance(df.index, pd.RangeIndex)
+        if config.get("CSV_EXPORT_USE_VERBOSE_NAMES_AS_HEADERS"):
+            d = {
+                o.column_name: o.verbose_name
+                for o in self.datasource.columns
+                if hasattr(o, "verbose_name")
+            }
+            replaced_columns = {
+                column_name: d.get(column_name)
+                for column_name in df.columns
+                if d.get(column_name)
+            }
```

Comment:
Is there some reason this is done in two steps? I wouöd think that it should be adequate to do dict comprehension on the `dd` columns and then just checking if those cols have a verbose name?

Reply:
Fixed, please take another look.

Reply:
What I was thinking was that as there is already a loop of all columns in `self.datasource.columns`, one could just add an `and`to the if of the first dict comprehension to check whether or not that column is already present in the dataframe. That would save the effort of doing yet another dict comprehension. Then the if part would look like this:
```python3
                if hasattr(column, "verbose_name") and column.verbose_name and column.column_name in df.columns
```
which would make it possible to skip the second dict comprehension and just do
```python3
            df = df.rename(columns=columns_with_verbose_names)
```

Reply:
Thank you for the guidance, your solution is indeed much clear